src/sentry/middleware/ratelimit.py

```diff
@@ -22,6 +22,8 @@
 class RatelimitMiddleware(MiddlewareMixin):
     """Middleware that applies a rate limit to every endpoint."""
 
+    rate_limit_check_dict: dict | None = None
```

Comment:
The type here would have to be `Mapping[str, bool | int] | None`. It also might make sense to make a dataclass just for the ratelimit payload.